src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -661,6 +661,73 @@ static void Main()
                 Diagnostic(ErrorCode.WRN_ImplicitCopyInReadOnlyMember, "P2").WithArguments("S.P2.get", "this").WithLocation(11, 17));
         }
 
+        [Fact]
+        public void ReadOnlyMethod_CallEventAccessor()
+        {
+            var csharp = @"
+using System;
+
+public struct S
+{
+    public readonly void M()
+    {
+            Console.WriteLine(E is null);
+            // should create local copy
+            E += () => {}; // warning
+            Console.WriteLine(E is null);
+            E -= () => {}; // warning
```

Comment:
>E -= () => {}; // warning [](start = 12, length = 25)

Consider also testing direct assignment: `E = () => { };`

Reply:
There's existing coverage for this. It's always an error within a readonly method. Will move event tests closer together so it's easier to find.